### ISSUE TITLE
Dataset processing details are missing: EDA_unlabelled, EDA_labelled (cannot perform contrastive pretraining)

### ISSUE THREAD

**Issue by sumeromer** [kmatton/contrastive-learning-for-eda#1]
I read the paper and wanted to train on WESAD; however, ```EDA_unlabelled.csv``` and ```EDA_labelled.csv``` files are missing.
I also noticed in the README file it is noted as TODO.

Instead of guessing what could be the content of these files from ```datasets//dataset_utils.py```, can you share processed files, a sample of these files, or the structure (i.e., x_left_buffer, x_right_buffer, x,  y, timestamps and so on). Without knowing these, running the code and replicating any results is not doable.

**Comment by sumeromer** [comment]
Still, the data structure of these CSV files is needed. I processed two datasets, WESAD and VerBIO:

In both datasets, the sampling rate is 4Hz. Also, from the paper:  
**"60-second windows, overlapping with a window shift of 0.25 seconds"** (it means shifting one item in raw EDA files, as they are sampled at 4Hz and taking 60x4=240 length sequences).

* In WESAD:
   - In the paper: 103,037 segments with non-stressed (65%) and stressed (35%).
   - What I calculated: 105,453 segments with non-stressed (65%) and stressed (35%).
   
 * In VerBIO:
    There are missing E4 EDA features in some samples. 
     - In the paper:  403,072 segments with non-stressed (39%) and stressed (61%).
     - I calculated: 292,757 segments with non-stressed (23%) and stressed (77%).
     
In WESAD, the difference is smaller, but why is there a huge difference in the VerBIO dataset?

If you do not share your preprocessing script or more detail about the input data structure, reproducing your approach will never give a similar result.

**Comment by kmatton** [comment]
Thank you for your interest in our code! 

Regarding the VerBIO dataset, the large deviation is likely because we received additional E4 dataset files from the dataset's authors. We will connect you with the authors over email, so you can also obtain these files.

We will also release the dataset processing code soon.

**Comment by kmatton** [comment]
We have now added our dataset preprocessing code and updated the README to explain how to run the code on the WESAD/VerBIO datasets. Hope that helps!

**Comment by sumeromer** [comment]
Hi Katie, thank you very much for your reply! I contacted the authors of the VerBIO dataset and acquired the additional training data, too. I replicated your supervised baseline in the following manner:

I pick random n=5 subjects of VerBIO as validation, the remaining as training set (n=50), and train the model max 200 epochs. Mostly early stopping stops at the end of 7-8 or max. 15 epochs. Then, I directly tested on WESAD without applying any fine-tuning process. As the selected random five validation subjects will affect the performance, I repeated five times (ACC=0.688±0.019, AUROC score=0.705±0.008). Even without using 1% randomly labeled data, WESAD performance is around 65-70%. 

However, again, I could not achieve exactly the same number of instances. In my processing, WESAD yielded 117,579, and VerBIO 351,741 instances when I applied as the way described in the paper. Now, I will use your scripts directly to get more consistent baseline results and replicate all experiments to compare fairly (i.e., all compared experiments with the same seeds).

Again, many thanks for the clarification and for providing the processing script!
